File: rust/advent2022/src/day_01.rs
pub fn go() {
    let mut cals: _ = include_str!("../inputs/day.01.input")
        .split("\n\n")
        .map(|x| x.split("\n").map(|x | x.parse::<u32>().unwrap()).sum())
        .collect::<Vec<u32>>();
    
    cals.sort_by(|x, y| x.cmp(y).reverse());
    
    println!("day 1 part 1: {}", cals[0]);
    println!("day 1 part 2: {}", cals[0] + cals[1] + cals[2]);
    
}

File: kotlin/advent2022/src/main/kotlin/Day01.kt
import java.nio.file.Files
import java.nio.file.Path
import java.util.stream.Collectors

class Stub {
    
}
fun main() {
    val stub = Stub()
    val cals = Files.readAllLines(Path.of(stub.javaClass.getClassLoader().getResource("day.01.input").toURI())).stream()
        .collect(Collectors.joining(",")).split(",,").stream()
        .map { x -> x.split(",")
                     .map { x -> Integer.parseInt(x) }
                     .reduce { acc, i -> acc + i } }
        .sorted(Comparator.reverseOrder()).toList()
    
    println("day 1 part 1: " + cals[0])
    println("day 1 part 2: " + (cals[0] + cals[1] + cals[2]))
}